src/Scanner.php

```diff
@@ -0,0 +1,266 @@
+  protected $entityFieldManager;
+
+  /**
+   * The logger service.
+   *
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $logger;
+
+  /**
+   * Source scanner constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\Entity\EntityTypeBundleInfoInterface $entity_type_bundle_info
+   *   The entity type bundle info service.
+   * @param \Drupal\Core\Entity\EntityFieldManager $entity_field_manager
+   *   The entity field manager.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   The logger service.
+   */
+  public function __construct(
+    EntityTypeManagerInterface $entity_type_manager,
+    EntityTypeBundleInfoInterface $entity_type_bundle_info,
+    EntityFieldManager $entity_field_manager,
+    LoggerInterface $logger) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->entityTypeBundleInfo = $entity_type_bundle_info;
+    $this->entityFieldManager = $entity_field_manager;
+    $this->logger = $logger;
+  }
+
+  /**
+   * Read directory with source CSV files.
+   *
+   * @param string $directory
+   *   Path to directory with source .csv files.
+   * @param array $csv_options
+   *   Additional options for the command.
+   *
+   * @return array
+   *   Array of source infos.
+   */
+  public function readSources($directory, array $csv_options) {
+    $sources = [];
+    // Read files from specified folder.
+    if (!is_dir($directory)) {
+      throw new \Exception('The destination directory does not exist.');
+    }
+    // Look for source files in defined directory.
+    $source_files = glob($directory . DIRECTORY_SEPARATOR . $csv_options['pattern']);
+    if (empty($source_files)) {
+      throw new \Exception('Source files not found in the destination directory.');
+    }
+
+    foreach ($source_files as $source_file) {
+      // Extract entity type, bundle from the file name. Supported:
+      // {entity_type}-{bundle}.csv or {entity_type}.csv.
+      preg_match('!
+        ([a-z_]+)             # entity type (group 1)
+        (
+          \\-                 # -
+          ([^\\./]+)          # bundle (group 3)
+        )?                    # (can be empty)
+        \\.                   # .
+        csv                   # csv extension
+      $!x', $source_file, $matches);
+      if (isset($matches[1])) {
+        $entity_type = $matches[1];
+        $bundle = $matches[3] ?: $entity_type;
+      }
+      else {
+        // Skip if cannot parse source file name.
+        $this->logger->alert(
+          'Cannot parse source filename %name. Skipping.',
+          ['%name' => $source_file]
+        );
+        continue;
+      }
+
+      $bundle_info = $this->entityTypeBundleInfo->getBundleInfo($entity_type);
+      // Skip if bundle not exists.
+      if (!($bundle_info && isset($bundle_info[$bundle]))) {
+        $this->logger->alert(
+          'Bundle %bundle not exists for %entity entity. Skipping.',
```

Comment:
If file gets skipped, this message should rather throw a warning instead of an error as it currently does : 

![image](https://user-images.githubusercontent.com/20738264/82908355-22f33200-9f68-11ea-8219-38b8bd50c249.png)

Reply:
fixed